railties/test/commands/temp_test.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "rails/command"
+
+class Rails::Command::TempTest < ActiveSupport::TestCase
+  setup :build_app
+  teardown :teardown_app
+
+  test "tmp:clear clear cache, socket, screenshot, and storage files" do
+    Dir.chdir(app_path) do
+      FileUtils.mkdir_p("tmp/cache")
+      FileUtils.touch("tmp/cache/cache_file")
+
+      FileUtils.mkdir_p("tmp/sockets")
+      FileUtils.touch("tmp/sockets/socket_file")
+
+      FileUtils.mkdir_p("tmp/screenshots")
+      FileUtils.touch("tmp/screenshots/fail.png")
+
+      FileUtils.mkdir_p("tmp/storage/6h/np")
+      FileUtils.touch("tmp/storage/6h/np/6hnp81jvgt42pcfqtlpoy8qshfb0")
+
+      rails "tmp:clear"
+
+      assert_not File.exist?("tmp/cache/cache_file")
+      assert_not File.exist?("tmp/sockets/socket_file")
+      assert_not File.exist?("tmp/screenshots/fail.png")
+      assert_not File.exist?("tmp/storage/6h/np/6hnp81jvgt42pcfqtlpoy8qshfb0")
+    end
+  end
+
+  test "tmp:create creates tmp folders" do
+    Dir.chdir(app_path) do
+      rails "tmp:create"
+      assert Dir.exist?("tmp/cache")
+      assert Dir.exist?("tmp/sockets")
+      assert Dir.exist?("tmp/pids")
+      assert Dir.exist?("tmp/cache/assets")
```

Comment:
I think it would be good to call `FileUtils.rm_rf(tmp_dirs)` before `rails "tmp:create"`.  That way we can be sure we are actually testing creation.

Reply:
Done.